File: auth-data-server/src/main/java/com/cartobucket/auth/rpc/server/entities/Event.java
package com.cartobucket.auth.rpc.server.entities;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.type.SqlTypes;

import java.time.OffsetDateTime;
import java.util.Map;
import java.util.UUID;

@Entity
public class Event {
    @Id
    @GeneratedValue
    private UUID id;
    private EventType eventType;
    private String version;
    private UUID authorizationServerId;
    private UUID resourceId;
    @JdbcTypeCode(SqlTypes.JSON)
    private Map<String, Object> resource;
    private OffsetDateTime createdOn;

    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    public EventType getEventType() {
        return eventType;
    }

    public void setEventType(EventType eventType) {
        this.eventType = eventType;
    }

    public String getVersion() {
        return version;
    }

    public void setVersion(String version) {
        this.version = version;
    }

    public UUID getAuthorizationServerId() {
        return authorizationServerId;
    }

    public void setAuthorizationServerId(UUID authorizationServerId) {
        this.authorizationServerId = authorizationServerId;
    }

    public UUID getResourceId() {
        return resourceId;
    }

    public void setResourceId(UUID resourceId) {
        this.resourceId = resourceId;
    }

    public Map<String, Object> getResource() {
        return resource;
    }

    public void setResource(Map<String, Object> resource) {
        this.resource = resource;
    }

    public OffsetDateTime getCreatedOn() {
        return createdOn;
    }

    public void setCreatedOn(OffsetDateTime createdOn) {
        this.createdOn = createdOn;
    }

}

File: auth-data-postgres-client/src/main/kotlin/com/cartobucket/auth/postgres/client/entities/Event.kt
package com.cartobucket.auth.postgres.client.entities

import com.cartobucket.auth.data.domain.ResourceType
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.Id
import org.hibernate.annotations.JdbcTypeCode
import org.hibernate.type.SqlTypes
import java.time.OffsetDateTime
import java.util.UUID

@Entity
class Event {
    @Id
    @GeneratedValue
    var id: UUID? = null

    var eventType: EventType? = null

    var version: String? = null

    var authorizationServerId: UUID? = null

    var resourceType: ResourceType? = null

    var resourceId: UUID? = null

    @JdbcTypeCode(SqlTypes.JSON)
    var resource: Map<String, Any?>? = null

    var createdOn: OffsetDateTime? = null
}
